lib/p2p/P2PRepository.ts

```diff
@@ -19,11 +19,19 @@ class P2PRepository {
   }
 
   public addHost = async (host: db.HostFactory): Promise<db.HostInstance> => {
-    return this.models.Host.upsert(<db.HostAttributes>host , { returning: true })[0];
+    return this.models.Host.findOne({ where: { address: host.address, port: host.port } }).then((result): Promise<db.HostInstance> => {
```

Comment:
It seems not ideal to have to hit the database to see if a host exists, considering that we load all hosts on startup. We have a `getOrCreateHost` method in HostList.ts that should only attempt to create a host if it's not already in the list. Can you see why that method. I think we should actually be able to leave this `addHost` method as it was before, simply doing an insert, and make sure not to even get to this point if we're attempting to insert a host that already exists.

Reply:
Another thought is that we might want to update a host row even if it exists with the new handshake data, maybe don't need to worry about it for this PR tho. The key though is that we should be able to tell if the host exists without needing to query the database.